### ISSUE TITLE
xenstat plugin doesn't build against Xen 4.13

### ISSUE THREAD

**Issue by stormi** [netdata/netdata#7803]
##### Bug report summary
The xenstat plugin, that compiled well against xen 4.11.1, fails to build against xen 4.13.

Here are the error messages:
```
collectors/xenstat.plugin/xenstat_plugin.c: In function 'xenstat_collect':
collectors/xenstat.plugin/xenstat_plugin.c:418:5: warning: implicit declaration of function 'xenstat_node_freeable_mb' [-Wimplicit-function-declaration]
     node_metrics.freeable_mb = xenstat_node_freeable_mb(node);
     ^
collectors/xenstat.plugin/xenstat_plugin.c:460:9: error: unknown type name 'xenstat_tmem'
         xenstat_tmem *tmem = xenstat_domain_tmem(domain);
         ^
collectors/xenstat.plugin/xenstat_plugin.c:460:9: warning: implicit declaration of function 'xenstat_domain_tmem' [-Wimplicit-function-declaration]
collectors/xenstat.plugin/xenstat_plugin.c:460:30: warning: initialization makes pointer from integer without a cast [enabled by default]
         xenstat_tmem *tmem = xenstat_domain_tmem(domain);
                              ^
collectors/xenstat.plugin/xenstat_plugin.c:461:9: warning: implicit declaration of function 'xenstat_tmem_curr_eph_pages' [-Wimplicit-function-declaration]
         d->tmem.curr_eph_pages = xenstat_tmem_curr_eph_pages(tmem);
         ^
collectors/xenstat.plugin/xenstat_plugin.c:462:9: warning: implicit declaration of function 'xenstat_tmem_succ_eph_gets' [-Wimplicit-function-declaration]
         d->tmem.succ_eph_gets = xenstat_tmem_succ_eph_gets(tmem);
         ^
collectors/xenstat.plugin/xenstat_plugin.c:463:9: warning: implicit declaration of function 'xenstat_tmem_succ_pers_puts' [-Wimplicit-function-declaration]
         d->tmem.succ_pers_puts = xenstat_tmem_succ_pers_puts(tmem);
         ^
collectors/xenstat.plugin/xenstat_plugin.c:464:9: warning: implicit declaration of function 'xenstat_tmem_succ_pers_gets' [-Wimplicit-function-declaration]
         d->tmem.succ_pers_gets = xenstat_tmem_succ_pers_gets(tmem);
         ^
  CC       daemon/common.o
collectors/xenstat.plugin/xenstat_plugin.c: In function 'main':
collectors/xenstat.plugin/xenstat_plugin.c:1189:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
make[2]: *** [collectors/xenstat.plugin/xenstat_plugin.o] Error 1
make[2]: *** Waiting for unfinished jobs....
collectors/perf.plugin/perf_plugin.c: In function 'main':
collectors/perf.plugin/perf_plugin.c:1348:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
make[2]: Leaving directory `/home/builder/rpmbuild/BUILD/netdata-1.19.0'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/builder/rpmbuild/BUILD/netdata-1.19.0'
make: *** [all] Error 2
```

Xen 4.13 is the version that is included in the upcoming XCP-ng 8.1 that I'm preparing as the main packager, and in Citrix Hypervisor 8.1.

The build environment is a container that can be replicated using https://github.com/xcp-ng/xcp-ng-build-env and https://github.com/xcp-ng-rpms/netdata/ but my assumption is that it will be easy to reproduce in your own environment if you update Xen.

Another assumption from me is that it is related to the version of Xen. In case I'm mistaking I'll dig deeper.

##### Component Name
xenstat_plugin

##### Steps To Reproduce
* Tested steps: build in XCP-ng's build container
* Assumed steps: build against Xen 4.13

**Comment by Ferroin** [comment]
Quick note to whoever ends up working on this:

4.13 is the most recent release of Xen (2019-12-18), and they don't use semantic versioning, so this is likely due to an API change upstream.

**Comment by amoss** [comment]
@vlvkobal Please take a look and see if this can be replicated.

**Comment by vlvkobal** [comment]
We need one more approval for #7951. @stelfrag could you look at it?

**Comment by vlvkobal** [comment]
Fixed in #7951.
